prereq_flowchart/scrape/scrape.py

```diff
@@ -0,0 +1,79 @@
+import os
+from os import path
+import json
+from pprint import pprint
+import pickle
+
+from prereq_flowchart.scrape import classinfo
+from prereq_flowchart.scrape import majorScrapper
+
+"""
+functions for collecting information from the web
+
+todo: select for specific info instead of gathering everything and then sorting 
+through it
+
+ultimately it would be nice to be able to request data for specific classes and 
+construct graphs with minimum viable data, which is why i extracted this logic
+out (in order to create an adapted interface to the websites with built-in 
+caching)
+"""
+
+# prereq_flowchart/prereq_flowchart/scrape/../../data
+DATA_FOLDER = path.join(path.dirname(path.abspath(__file__)), "../../data")
+if not path.exists(DATA_FOLDER):
+    os.makedirs(DATA_FOLDER)
+FORCE_RECHECK_CLASSINFO = False
+FORCE_RECHECK_MAJORINFO = False
+
+
+def scrape_classinfo() -> dict[classinfo.CourseNumber, classinfo.Course]:
+    pickle_path = path.join(DATA_FOLDER, "out_classinfo.pickle")
+    log_path = path.join(DATA_FOLDER, "out_classinfo.txt")
+
+    if path.exists(pickle_path) and not FORCE_RECHECK_CLASSINFO:
```

Comment:
feels like `FORCE_RECHECK_CLASSINFO` should be an argument that defaults to false?

Reply:
tbh it was just for debugging, and i didn't even use it (just deleted the data folder), but i think i will bring it up to main as a flag

i also sorta wanna make more specific requests to collect minimal data but that's in the future